src/utils/createAndAwaitJob.ts

```diff
@@ -0,0 +1,41 @@
+export default async function createAndAwaitJob<
+  Job extends { done?: boolean; error?: any; errorMsg?: any }
+>(
+  fetcherWithSign: ([string, any]) => Promise<any>,
+  url: string,
+  body: any,
+  queryParams: Record<string, any>
+) {
+  const poll = (): Promise<Job[]> =>
+    fetcherWithSign([
+      `${url}?${new URLSearchParams(queryParams).toString()}`,
+      { method: "GET" },
+    ])
+
+  const initialPollResult = await poll().catch(() => null as Job[])
+
+  const jobAlreadyInProgress = initialPollResult?.find((job) => !job.done)
```

Comment:
Will this work properly? I mean what if a job is in progress, but the user starts another job (e.g. another join flow)? Shouldn't we distinguish jobs somehow? 👀 

(btw it's good for now I think, since we'll only enable this feature in one guild)

Reply:
Yes, I think it'll be fine this way then, thanks!